tests/example_integration.rs

```diff
@@ -0,0 +1,194 @@
+use bevy::prelude::*;
+use bevy_ggrs::{GGRSPlugin, GGRSSchedule, PlayerInputs, Rollback, RollbackIdProvider, Session};
+use ggrs::{Config, SessionBuilder, PlayerType, UdpNonBlockingSocket, PlayerHandle, P2PSession};
+
+#[test]
+#[serial]
+fn it_syncs_rollback_resources() -> Result<(), Box<dyn std::error::Error>> {
+    let (player1, player2) = create_players();
+    let session1 = start_session(&player1, &player2)?;
+    let mut app1 = create_app::<GGRSConfig>(session1);
+    let session2 = start_session(&player2, &player1)?;
+    let mut app2 = create_app::<GGRSConfig>(session2);
+
+    for _ in 0..50 {
+        app1.update();
+        app2.update();
+    }
+
+    let frame_count1 = app1.world.get_resource::<FrameCount>().unwrap();
+    let frame_count2 = app2.world.get_resource::<FrameCount>().unwrap();
+    assert!(frame_count1.frame > 0);
+    assert!(frame_count2.frame > 0);
+    assert_eq!(frame_count1.frame, frame_count2.frame);
+    Ok(())
+}
+
+#[test]
+#[serial]
+fn it_syncs_rollback_components() -> Result<(), Box<dyn std::error::Error>> {
+    let (player1, player2) = create_players();
+    let session1 = start_session(&player1, &player2)?;
+    let mut app1 = create_app::<GGRSConfig>(session1);
+    let session2 = start_session(&player2, &player1)?;
+    let mut app2 = create_app::<GGRSConfig>(session2);
+
+    for _ in 0..250 {
+        press_key(&mut app1, KeyCode::W);
+        app1.update();
+        app2.update();
+    }
+
+    let mut app2_query = app2.world.query::<(&Transform, &PlayerComponent)>();
+    for (transform, player) in app2_query.iter(&app2.world) {
+        if player.handle == player1.handle {
+            assert!(transform.translation.z < 0., "Remote player moves forward");
+        }
+    }
+    Ok(())
+}
+
+pub struct TestPlayer {
+    handle: PlayerHandle,
+    address: SocketAddr,
+}
+
+fn create_players() -> (TestPlayer, TestPlayer) {
+    const PLAYER1_PORT: u16 = 8081;
+    const REMOTE_PORT: u16 = 8082;
+    let remote_addr1 = SocketAddr::new(IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)), PLAYER1_PORT);
+    let remote_addr2 = SocketAddr::new(IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)), REMOTE_PORT);
+    return (TestPlayer { handle: 0, address: remote_addr1 }, TestPlayer { handle: 1, address: remote_addr2 });
+}
+
+fn start_session(local_player: &TestPlayer, remote_player: &TestPlayer)
+                 -> Result<P2PSession<GGRSConfig>, Box<dyn std::error::Error>> {
+    let mut session_builder = SessionBuilder::<GGRSConfig>::new()
+        .with_num_players(2)
+        .with_max_prediction_window(12) // (optional) set max prediction window
+        .with_input_delay(2); // (optional) set input delay for the local player
+    session_builder = session_builder.add_player(PlayerType::Local, local_player.handle)?;
+    session_builder = session_builder.add_player(PlayerType::Remote(remote_player.address), remote_player.handle)?;
+    let socket = UdpNonBlockingSocket::bind_to_port(local_player.address.port())?;
```

Comment:
I guess we could even create a dummy in-memory socket if we want to make the tests able to run in parallel? Or simulate packet loss etc.

Reply:
Yeah this ended up being a end-to-end test scenario with the actual socket in use.

Ideally it would be a mocked socket session for as u mention, reusability and run them in parallel.

I'll do a redo of the integration test based on https://github.com/gschup/bevy_ggrs/pull/52